### ISSUE TITLE
Dynamodb listener not paginating through results of list_tables(), incorrectly making DescribeTable fail

### ISSUE THREAD

**Issue by psverkada** [localstack/localstack#2919]
# Type of request: This is a ...

[x ] bug report
[ ] feature request

# Detailed description

We use localstack for dynamodb.  When we create more than 100 dynamodb tables, random tables will start having operations fail on DescribeTables.  Like this fail of PutItem:

```
vusers_1                 | botocore.exceptions.ClientError: An error occurred (502) when calling the PutItem operation (reached max retries: 9): Bad Gateway
```

At the same time this error is happening, we are seeing this error in our localstack container:

```
localstack_1             \| 2020-09-02T00:56:23:ERROR:localstack.services.generic_proxy: Error forwarding request: An error occurred (ResourceNotFoundException) when calling the DescribeTable operation: Cannot do operations on a non-existent table Traceback (most recent call last):
--
73 | localstack_1             \|   File "/opt/code/localstack/localstack/services/generic_proxy.py", line 240, in forward
74 | localstack_1             \|     path=path, data=data, headers=forward_headers)
75 | localstack_1             \|   File "/opt/code/localstack/localstack/services/dynamodb/dynamodb_listener.py", line 53, in forward_request
76 | localstack_1             \|     ProxyListenerDynamoDB.thread_local.existing_item = find_existing_item(data)
77 | localstack_1             \|   File "/opt/code/localstack/localstack/services/dynamodb/dynamodb_listener.py", line 337, in find_existing_item
78 | localstack_1             \|     schema = ddb_client.describe_table(TableName=table_name)
79 | localstack_1             \|   File "/opt/code/localstack/.venv/lib/python3.6/site-packages/botocore/client.py", line 357, in _api_call
80 | localstack_1             \|     return self._make_api_call(operation_name, kwargs)
81 | localstack_1             \|   File "/opt/code/localstack/.venv/lib/python3.6/site-packages/botocore/client.py", line 661, in _make_api_call
82 | localstack_1             \|     raise error_class(parsed_response, operation_name)
83 | localstack_1             \| botocore.errorfactory.ResourceNotFoundException: An error occurred (ResourceNotFoundException) when calling the DescribeTable operation: Cannot do operations on a non-existent table
```

The cause seems to be that in the DescribeTables handler (and every other place list_tables() is called) in dynamodb_listener.py, only the first page, which by default is 100 tables, is used:

```
        elif action == '%s.DescribeTable' % ACTION_PREFIX:
            # Check if table exists, to avoid error log output from DynamoDBLocal
            table_names = ddb_client.list_tables()['TableNames']
            if to_str(data['TableName']) not in table_names:
                return get_table_not_found_error()
```

Botocore doc on list_tables:

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodb.html#DynamoDB.Client.list_tables

>  The output from ListTables is paginated, with each page returning a maximum of 100 table names.

...

## Expected behavior

The PutItem operation should not 502 - the table was created successfully so it should exist.
...

## Actual behavior

The PuItem operation fails with a 502.

...

# Steps to reproduce

1. Create more than 100 dynamob tables with localstack
2. Try to do a PutItem on each table - you should get (N - 100) 502 errors, where N is the number of tables.

## Command used to start LocalStack

We launch the container using docker-compose.  We are building a container off of your localstack:0.10.5 container, and just adding some init scripts to create various resources.

Here is the docker-compose yml for it:

```
  localstack:
    build:
      dockerfile: docker/localstack/Dockerfile
      context: .
    environment:
      <<: *common-variables
      SERVICES: cloudwatch,s3,sqs,lambda,dynamodb,kinesis,elasticsearch,sns
      DEBUG: ${DEBUG- }
      DATA_DIR: /tmp/localstack/data
      PORT_WEB_UI: ${PORT_WEB_UI- }
      LAMBDA_EXECUTOR: docker
      LAMBDA_DOCKER_NETWORK: host
      KINESIS_ERROR_PROBABILITY: ${KINESIS_ERROR_PROBABILITY- }
      DOCKER_HOST: unix:///var/run/docker.sock
      HOSTNAME_EXTERNAL: localhost
      HOSTNAME: localstack
      AWS_ACCESS_KEY_ID: localstack
      AWS_SECRET_ACCESS_KEY: localstack
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock # allow lambda to start docker containers
    ports:
      - 8888

```
...

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

Here's how we create tables:

```
    dynamo = boto3.client('dynamodb', endpoint_url=endpoint_url, region_name='us-west-2')

    def create_table(name, table):
        print('creating %s' % name)
        try:
            dynamo.create_table(**table)
        except ClientError as e:
            if e.response['Error']['Code'] == 'ResourceInUseException':
                print('=> already created')
                pass
            else:
                raise

```

Here's the failing call to put_item:

```
        self.dynamo_resource = AWS.resource("dynamodb", region_name=self.get_config("REGION"))
        self.user_metadata_table = self.dynamo_resource.Table(user_metadata_table_name)
...
        self.user_metadata_table.put_item(Item=item)


```
...

**Comment by psverkada** [comment]
This bug is breaking our master builds for us, since we use localstack for CI.  So would be great to get a fix for this quickly, or happy to submit a PR ourselves.  Let me know.  Thanks!

**Comment by markddavidoff** [comment]
@usmangani1 here is my suggestion:
https://github.com/localstack/localstack/pull/2923

**Comment by psverkada** [comment]
Hi, any updates on this @usmangani1 ?

**Comment by whummer** [comment]
Hi @psverkada , looks like this should already be fixed via  #2923 - can you please give it another try after pulling the `latest` Docker image? Please report here if the problem persists. Thanks
